languages/purescript/reference/concepts/modules.md

```diff
@@ -0,0 +1,185 @@
+# Modules
+
+All code in PureScript is contained in a module. Modules are introduced using the `module` keyword:
+
+```purescript
+module A where
+  
+id x = x
+```
+
+## Importing Modules
+
+A module can be imported using the `import` keyword. This is called an "open import" - it will create aliases for all of the values and types in the imported module:
+
+```purescript
+module B where
+  
+import A
+```
+
+Alternatively, a list of names to import can be provided in parentheses:
+
+```purescript
+module B where
+  
+import A (runFoo)
+```
+
+Values, operators (wrapped in parentheses), type constructors, data constructors, and type classes can all be explicitly imported. A type constructor can be followed by a list of associated data constructors to import in parentheses. A double dot (`..`) can be used to import all data constructors for a given type constructor:
+
+```purescript
+module B where
+
+import A (runFoo, (.~), Foo(..), Bar(Bar))
```

Comment:
Perhaps a well known operator like `+` would make the syntax a little more obvious